File: ruby/bob/bob.rb
class Bob
  def numeric?
    return true if self =~ /\A\d+\Z/
    true if Float(self) rescue false
  end

  def self.hey(phrase)
    phrase = phrase.tr("0-9", "")
    if phrase == phrase.upcase
      return 'Whoa, chill out!'
    elsif phrase.include?('?')
      return 'Sure.'
    else
      return 'Whatever.'
    end
  end
end

File: javascript/bob/bob.js
var Bob = function() {};

Bob.prototype.hey = function(phrase) {
  let output = "Whatever.";
  if (phrase === phrase.toUpperCase() || phrase.to_i) {
    output = "Whoa, chill out!";
  } else if (phrase.includes("?")) {
    output = "Sure.";
  }
  return output;
};

module.exports = Bob;
